File: Frontend/src/app/Domain/appUser.ts
export class AppUser {
  id: number = 0;
  firstName: string = '';
  lastName: string = '';
  phoneNumber: string = '';
  email: string = '';
  amountPaid: number = 0;
  amountToBePaid: number = 0;
}

export class CreateAppUserDto {
  firstName: string = '';
  lastName: string = '';
  phoneNumber: string = '';
  email: string = '';
}

File: Api/Domain/Models/AppUser.cs
namespace Domain.Models
{
    public class AppUser : BaseObject
    {
        public required string FirstName { get; set; }
        public required string LastName { get; set; }
        public required string PhoneNumber { get; set; }
        public required string Email { get; set; }
        public int AmountPaid { get; set; } = 0;
        public int AmountToBePaid { get; set; } = 0;
        public IEnumerable<RoomReservation> RoomReservations { get; set; } = new List<RoomReservation>();
        public IEnumerable<CourseUserAssigment> CourseUserAssigment { get; set; } = new List<CourseUserAssigment>();
    }
}
